### ISSUE TITLE
autosuggestion

### ISSUE THREAD

**Issue by uGeek** [vonshednob/pter#6]
It would be possible that when you write +, @, due: ... all the options to be selected appear, to register tasks faster and not make mistakes when writing them.

Very good application. I like very much

Thank you

**Comment by vonshednob** [comment]
Oh yeah, I absolutely agree! Currently there’s a bigger refactoring on-going which will make this easier.
So, it’ll happen, but not very soon.

Not quite sure about the `due:` completion though; since it’s very dynamic what you could write there (`2022-02-02`, `+4d`, `tue`, `+1y2w-2d`) it would almost require a calendar selection, but even that wouldn’t quite cut it.
Any ideas?

**Comment by maxigaz** [comment]
One is idea is to make an option to specify the date format, and use it as the basis of converting to/from.

In [khal](https://lostpackets.de/khal/), for example, you can set the format for both the short date and the long date. I have the following in my `~/.config/khal/config`:

```
[locale]
timeformat = %H:%M
dateformat = %m.%d.
longdateformat = %Y-%m-%d
datetimeformat = %Y-%m-%d %H:%M
longdatetimeformat = %Y-%m-%d %H:%M
```

So, maybe pter could detect that if I started writing `due:2021` and I have `%Y-%m-%d` configured as the format, so pressing tab could complete the date as `due:2021-10-17`.

**Comment by vonshednob** [comment]
Not quite sure how it would complete the date though, if I only type `due:2021` and Tab… would it show a completion option for month and day?

Just curious: are you using absolute dates in `due:` and `t:` more often than relative dates? I keep finding myself rather settings tasks up to be at `due:sat` or `due:+1w`.

**Comment by vonshednob** [comment]
Autocompletion is implemented in [2.1.0](https://github.com/vonshednob/pter/blob/master/CHANGELOG.md#210).

I was a bit too eager to push it out of the door, so the documentation is not great (i.e. missing). It’ll come later.

For now: Tab, `^N`, and arrow key down will select the next available option, `^P`, and arrow key up will select the previous option. `Enter` is used to select the completion option, `Esc` and `^C` will close the completion box.

Completion works for contexts (`@`) and projects (`+`).

**Comment by maxigaz** [comment]
Thank you very much!
